### PR TITLE
Http middleware downgrade

### LINKED ISSUE #148: Installation is broken
After #143 `http-interop/http-middleware` was upgraded to `0.5.0` which is not yet supported by `zendframework/zend-expressive` (https://github.com/zendframework/zend-expressive/issues/513).

### PR DESCRIPTION
**Pull request by bgaleotti**
Fixes #148

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
         "php": "^7.1",
         "aura/router": "^3.0",
         "doctrine/dbal": "^2.5.5",
-        "http-interop/http-middleware": "^0.5.0",
+        "http-interop/http-middleware": "^0.4.1",
```
Comment (sandrokeil) [inline composer.json]: What is the problem with version 0.5.0?

**Comment by bgaleotti** [comment]
Even if our code is compatible with "^0.5", `zendframework/zend-expressive` still does not support it ([unreleased](https://github.com/zendframework/zend-expressive/pull/517)).

Tried pinning “^0.2 | ^0.3 | ^0.4 | ^0.5” and "^0.5" gets installed.

**Comment by basz** [comment]
What does ‘composer why http-interop/http-middleware‘ tell you?

**Comment by basz** [comment]
Also don’t see that pinning in the PR

**Comment by bgaleotti** [comment]
Pinned `http-interop/http-middleware` in cf136f1995f45b87675e353895fc87f314898574

### Before:

```bash
> composer why http-interop/http-middleware

prooph/proophessor-do                     dev-http-middleware-downgrade  requires  http-interop/http-middleware (^0.4.1)
webimpress/http-middleware-compatibility  0.1.4                          requires  http-interop/http-middleware (^0.1.1 || ^0.2 || ^0.3 || ^0.4.1 || ^0.5)
zendframework/zend-expressive             2.0.5                          requires  http-interop/http-middleware (^0.4.1)
```

### After:

```bash
> composer why http-interop/http-middleware

prooph/proophessor-do                     dev-http-middleware-downgrade  requires  http-interop/http-middleware (^0.2 | ^0.3 | ^0.4 | ^0.5)
webimpress/http-middleware-compatibility  0.1.4                          requires  http-interop/http-middleware (^0.1.1 || ^0.2 || ^0.3 || ^0.4.1 || ^0.5)
```

**Comment by basz** [comment]
Uhm why would expressive doesn’t show up in the second one, I’m confused. And not behind machine to unconfuse me... will have to check at a later time...

**Comment by basz** [comment]
It's a thing with the priority of dependency resolving. Or something...

If we require `"http-interop/http-middleware": "*"` and `"zendframework/zend-expressive": "^2.0"` instead of `zendframework/zend-expressive:2.0.5` we will get `2.0.4` which doesn't have the middleware pinned and we get "http-interop/http-middleware:0.5.0". If it would resolve to `zendframework/zend-expressive:2.0.5` first it would install `http-interop/http-middleware:0.4.1` because that is pinned.

So we should either pin it to 0.4.1 too and wait for zend to merge te PR remove our `http-interop/http-middleware` at all. In testing I found that I get `zendframework/zend-expressive:2.0.5` with `http-interop/http-middleware:0.4.1` with both methods. 

So, I think we should remove the dependency altogether as it is required via `webimpress/http-middleware-compatibility`.

**Comment by prolic** [comment]
@codeliner can you check this as well?

**Comment by basz** [comment]
I had a look on this again. TL;DR; we should pin to 0.4.1 until zend's PR is pulled.

Remember `webimpress/http-middleware-compatibility` comes with a plugin that will ask to install a version of `http-interop/http-middleware` if it isn't present in composer. Travis gets its vendor from its cache so it isn't asked there.

So, a user has two choices, latest or an earlier version.

As he will most likely choose the latest version, we will end up with an older version of `zend-expressive`.

When he knows about this dependency dance, and chooces to install version 0.4 of `http-interop/http-middleware` he will end up with the correct version of zend-expressive. I guess the plugin only looks at root dependencies...

So, In the end I think we should pin the dependency to `http-interop/http-middleware:0.4.1` until zend is updated. And I'll post an issue with the plugin to see if this can be mitigated.

**Comment by basz** [comment]
sorry for the confusion...

**Pull request merged**